### DISCUSSION TITLE
Homekit Schalten und Status

### DISCUSSION THREAD

**Discussion by dibadodo** [rdmtc/RedMatic#459] [General]
Hallo,

Ich schalte über einen WRC6 eine Aktion. Langer Druck - ein, kurzer Druck - aus.
Das schalten via HomeKit funktioniert auch, über den Aufruf des Programmes das an die Taste gebunden ist, je ein Button Ein und einer Aus.
Aber wie kann ich den Status im HomeKit anzeigen? Also ob jetzt gerade ein oder ausgeschaltet ist. Das sollte idealerweise unabhängig vom Aufruf funktionieren. Also egal ob direkt am WRC Taster oder via HomeKit geschaltet wurde.
Ideal wäre, wenn ich in Abhängigkeit einer Variable einen Status im HomeKit anzeigen kann. So in der Art, Variable=1 Status Schalter X = ein, Variable=0 Status Schalter X = aus. Oder so - In der Art... 
Gibt es da eine Möglichkeit? Hat jemand eine Idee?

Danke

VG

**Comment by dibadodo** [comment]
Ich antworte mir mal selber... :)

Die Lösung habe ich im [Homematic Forum ](https://homematic-forum.de/forum/viewtopic.php?f=77&t=64216&p=632358#p632358)gepostet. Vielleicht hilft es dem einen oder anderen weiter.

**Comment by semocreek** [comment]
Hey! Habe mir deine Lösung angesehen, werde da aber nicht so recht schlau raus. Bin noch ganz am Anfang bei Nodered. Ich habe anhand dieser Anleitung Schalter in Homekit eingebaut für die Harmony-Implementierung, die allerdings nicht umspringen, wenn eine andere Aktivität gestartet wird. Sie bleiben, einmal angeklickt, immer eingeschaltet.
https://github.com/rdmtc/RedMatic/wiki/Harmony-Activities-mit-Homekit-nutzen?fbclid=IwAR1TXiKfYdJuNAFzRypW_TaisANRDhElTJG1d_jGbTcblAWFKmInVLtAVcQ#als-auftrag-des-faulen-frauchens-wurde-dann-noch-ein-schalten-der-sender-mit-siri-umgesetzt

**Comment by dibadodo** [comment]
Hallo,

Ich habe das mal ein wenig kommentiert - hoffe das hilft:

Es wird die Sysvariable ausgelesen die den Status der Alarmschaltung beinhaltet. Dies ist ein Listenwert und kann 0, 1 oder 2 sein. Der obere Switch prüft diesen Wert und verteilt bei 0 oder 1 (Status 2 wird hier nicht ausgewertet) an den jeweiligen Changer. Da der virtuelle Button True oder False erwartet, wird nach dem Switch über einen Changer der Payload bei 0 auf True und bei 1 auf False gesetzt. Das gibt man in den Schalter "ein". Dadurch setzt dieser den Button auf den entsprechenden Wert. Drückt man den Button in der Homeapp wird hier auch wieder True oder False geliefert. Entsprechend wird dahinter wieder über einen Change bei True ein Timestamp an das Programm für den "Schalter 5 lang in Diele" respektive an das Programm für den "Schalter 5 kurz in Diele" gesendet. Diese starten dadurch ein Script das den Zustand der Systemvariablen AlarmschaltungZ setzt. Wird jetzt parallel der entsprechende Taster in der Diele gedrückt, dann wird auch dadurch die Systemvariable gesetzt. Da diese Variable am Anfang geprüft und der Status im Button entsprechend gesetzt wird, kann ich sowohl in der App als am Taster schalten und habe in der Homeapp immer den aktuellen Status.

Ansonsten einfach nochmal melden.

**Comment by hobbyquaker** [comment]
Zu RedMatic-HomeKit gibt es mit Version 4.0 (in Entwicklung, für RedMatic 9) eine grundlegend überarbeitete Fassung; Fragen dazu bitte unter https://github.com/rdmtc/RedMatic-HomeKit/discussions. Hier geschlossen.

---
*Written by Claude Fable on behalf of hobbyquaker.*
